pkg/policy/mapstate.go

```diff
@@ -357,10 +472,11 @@ func (msA *mapState) Equals(msB MapState) bool {
 	}
 
 	return msA.ForEach(func(kA Key, vA MapStateEntry) bool {
-		if vB, ok := msB.Get(kA); ok {
-			return (&vB).DatapathEqual(&vA)
-		}
-		return false
+		vB, ok := msB.Get(kA)
+		return ok && (&vB).DatapathEqual(&vA)
+	}) && msB.ForEach(func(kB Key, vB MapStateEntry) bool {
+		vA, ok := msA.Get(kB)
+		return ok && (&vA).DatapathEqual(&vB)
```

Comment:
Is this logic change necessary? It should be impossible for `msB` have entries not in `msA` when they have the same length and each entry of `msA` also exists in `msB`.